inc/features/blocks/personalization/edit.js

```diff
@@ -0,0 +1,351 @@
+							className: 'altis-add-variant-button',
+							onClick: () => {
+								const newClientId = addVariant();
+								setVariant( newClientId );
+							},
+						},
+					] }
+				>
+					{ variants.map( variant => (
+						<Button
+							key={ `variant-select-${ variant.clientId }` }
+							className={ `altis-variant-button components-icon-button has-text ${ activeVariant === variant.clientId && 'is-active' }` }
+							title={ __( 'Select variant', 'altis-experiments' ) }
+							onClick={ () => setVariant( variant.clientId ) }
+						>
+							<VariantTitle variant={ variant } />
+						</Button>
+					) ) }
+				</Toolbar>
+			</BlockControls>
+			<InspectorControls>
+				{ variants.map( variant => {
+					if ( variant.attributes.fallback ) {
+						return (
+							<PanelBody
+								key={ `variant-settings-${ variant.clientId }` }
+								title={ __( 'Fallback', 'altis-experiments' ) }
+							>
+								<p className="description">
+									{ __( 'This variant will be shown as a fallback if no audiences are matched. You can leave the content empty if you do not wish to show anything.', 'altis-experiments' ) }
+								</p>
+							</PanelBody>
+						);
+					}
+
+					return (
+						<PanelBody
+							key={ `variant-settings-${ variant.clientId }` }
+							title={ <VariantTitle variant={ variant } /> }
+						>
+							<AudiencePicker
+								label={ __( 'Audience' ) }
+								audience={ variant.attributes.audience }
+								onSelect={ audience => {
+									setVariantAttributes( variant.clientId, { audience: audience.id } );
+								} }
+								onClearSelection={ () => {
+									setVariantAttributes( variant.clientId, { audience: null } );
+								} }
+							/>
+							{ ! variant.attributes.audience && (
+								<p className="description">
+									{ __( 'You must select an audience for this variant.', 'altis-experiments' ) }
+								</p>
+							) }
+						</PanelBody>
+					);
+				} ) }
+			</InspectorControls>
+			<style dangerouslySetInnerHTML={ {
+				__html: `
+					[data-block="${ clientId }"] [data-type="altis/personalization-variant"] {
+						display: none;
+					}
+					[data-block="${ clientId }"] #block-${ activeVariant } {
+						display: block;
+					}
+				`,
+			} } />
+			<div className={ className }>
+				<div className="wp-core-ui altis-experience-block-header">
+					<span className="altis-experience-block-header__title">
+						{ __( 'Personalized Content', 'altis-experiments' ) }
+						{ ' ・ ' }
+						<VariantTitle variant={ variants[ activeVariantIndex ] } />
+					</span>
+					{ isSelected && (
+						<div className="altis-experience-block-header__toolbar">
+							<IconButton
+								icon='migrate'
+								title={ __( 'Copy variant', 'altis-experiments' ) }
+								disabled={ variants.length < 1 }
+								onClick={ () => {
+									const newClientId = copyVariant( activeVariant );
+									setVariant( newClientId );
+								} }
```

Comment:
Can simplify to one line:

```suggestion
								onClick={ () => setVariant( copyVariant( activeVariant ) ) }
```